src/operations/common/resourceManager.js

```diff
@@ -0,0 +1,85 @@
+const globals = require('../../globals');
+const {AUDIT_EVENT_CLIENT_DB, ATLAS_CLIENT_DB, CLIENT_DB} = require('../../constants');
+const {getOrCreateCollection} = require('../../utils/mongoCollectionManager');
+
+/**
+ * returns the collection name for resourceType
+ * @param {string} resourceType
+ * @param {string} base_version
+ * @returns {string}
+ */
+function getCollectionNameForResourceType(resourceType, base_version) {
+    console.assert(!resourceType.endsWith('4_0_0'), `resourceType ${resourceType} has an invalid postfix`);
+    return `${resourceType}_${base_version}`;
+}
+
+/**
+ * returns the collection name for resourceType
+ * @param {string} resourceType
+ * @param {string} base_version
+ * @returns {string}
+ */
+function getHistoryCollectionNameForResourceType(resourceType, base_version) {
+    console.assert(!resourceType.endsWith('_History'), `resourceType ${resourceType} has an invalid postfix`);
+    return `${resourceType}_${base_version}_History`;
+}
+
+/**
+ * Gets the database connection for the given collection
+ * @param {string} resourceType
+ * @param {boolean?} useAtlas
+ * @returns {import('mongodb').Db}
+ */
+function getDatabaseConnectionForResourceType(resourceType, useAtlas) {
+    return (resourceType === 'AuditEvent') ?
+        globals.get(AUDIT_EVENT_CLIENT_DB) : (useAtlas && globals.has(ATLAS_CLIENT_DB)) ?
+            globals.get(ATLAS_CLIENT_DB) : globals.get(CLIENT_DB);
```

Comment:
Do we still need logic around CLIENT_DB? I thought this was the old DocDB which has since been deprecated?

Reply:
Sounds good, I wrote that issue up a little while ago:
https://icanbwell.atlassian.net/jira/software/c/projects/FSI/issues/FSI-36